libursa/DatabaseHandle.cpp

```diff
@@ -8,27 +8,26 @@ DatabaseHandle::DatabaseHandle(zmq::socket_t *worker) : worker(worker) {}
 
 [[nodiscard]] bool DatabaseHandle::request_dataset_lock(
     const std::vector<std::string> &ds_names) const {
-    s_send_val<NetAction>(*worker, NetAction::DatasetLockReq, ZMQ_SNDMORE);
+    s_send<NetAction>(worker, NetAction::DatasetLockReq, ZMQ_SNDMORE);
 
     for (const auto &ds_name : ds_names) {
-        s_send(*worker, "", ZMQ_SNDMORE);
-        s_send(*worker, ds_name, ZMQ_SNDMORE);
+        s_send_padding(worker, ZMQ_SNDMORE);
```

Comment:
random thought - maybe abstract `ZMQ_SNDMORE` even more?

SNDMORE just means `this is not the last message in this packeg`, There are probably million better ways to implement this in C++, for exmple:

```cpp
auto conn = s_begin();
conn.send("...");
conn.padding();
conn.send("qwe");
conn.padding();
conn.eof();
```

But ok, this is a pretty big change. What about abstracting this just a big at least:

```cpp
s_send(worker, "qwe");
s_send_padding(worker);
s_send(worker, "asfd");
s_send_padding(worker, SendFlags::Eof);
```

If this is the final form of the networking code I think it's worth it. Unless you want to refactor it even more in the future?

Thoughts?

Reply:
Yes, I'd like to gradually transition into something like the first example, so I'll leave it for now.

Reply:
:+1: